path/posix/extname.ts

```diff
@@ -18,26 +18,37 @@ import { isPosixPathSeparator } from "./_util.ts";
  * assertEquals(extname("/home/user/Documents/image.png"), ".png");
  * ```
  *
- * @example Working with URLs
+ * @param path The path to get the extension from.
+ * @returns The extension (ex. for `file.ts` returns `.ts`).
+ */
+export function extname(path: string): string;
+/**
+ * Return the extension of the `path` with leading period.
  *
- * Note: This function doesn't automatically strip hash and query parts from
- * URLs. If your URL contains a hash or query, remove them before passing the
- * URL to the function. This can be done by passing the URL to `new URL(url)`,
- * and setting the `hash` and `search` properties to empty strings.
+ * Note: Hashes and query parameters are ignore when constructing a URL.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @example Usage
  *
  * ```ts
  * import { extname } from "@std/path/posix/extname";
  * import { assertEquals } from "@std/assert";
  *
- * assertEquals(extname("https://deno.land/std/path/mod.ts"), ".ts");
- * assertEquals(extname("https://deno.land/std/path/mod.ts?a=b"), ".ts?a=b");
- * assertEquals(extname("https://deno.land/std/path/mod.ts#header"), ".ts#header");
+ * assertEquals(extname(new URL("https://deno.land/std/path/mod.ts")), ".ts");
+ * assertEquals(extname(new URL("https://deno.land/std/path/mod.ts?a=b")), ".ts");
+ * assertEquals(extname(new URL("https://deno.land/std/path/mod.ts#header")), ".ts");
  * ```
  *
  * @param path The path to get the extension from.
  * @returns The extension (ex. for `file.ts` returns `.ts`).
  */
-export function extname(path: string): string {
+export function extname(path: URL): string;
+export function extname(path: string | URL): string {
+  if (path instanceof URL) {
+    path = path.pathname;
```

Comment:
Can you use `fromFileUrl` instead? This allows other random urls such as `http:`, `data:`, `jsr:`, `node:`, etc, which is not aligned to how `dirname` works with those url types.

Reply:
What's the harm of using `.pathname`?

Reply:
In `data:` scheme, the pathname doesn't represent the path, but it represent arbitrary data. The return value of `extname` can be any random value in that case.